File: frontend/src/types/health.ts
export interface HealthResponse {
  status: string;
  app_name: string;
  version: string;
  environment: string;
  database_status: string;
  timestamp: string;
  cors_allowed: boolean;
}

export interface ApiState<T> {
  data: T | null;
  loading: boolean;
  error: string | null;
}

File: app/schemas/health.py
from pydantic import BaseModel, Field

class HealthCheckResponse(BaseModel):
    status: str = Field(..., example="ok")
    app_name: str = Field(..., example="Smart Queue Management Backend")
    version: str = Field(..., example="1.0.0")
    environment: str = Field(..., example="development")
    database_status: str = Field(..., example="connected")
    timestamp: str = Field(...)
    cors_allowed: bool = Field(default=True)
